yourgov/components/FormMobileFoodVendorPermit/task/FormStep7.tsx

```diff
@@ -1,72 +1,210 @@
+
+	const addedEmployee = query.success
+		? step7State?.employee &&
+		  step7State.employee.find((employee) => employee?.id === query.success)
+		: undefined;
+	const [showAddedEmployeeMessage, setShowAddedEmployeeMessage] = useState(
+		!!addedEmployee
+	);
+	useEffect(() => {
+		setShowAddedEmployeeMessage(!!addedEmployee);
+	}, [addedEmployee]);
+
+	const closeModal = () => {
+		setShowAddedEmployeeMessage(false);
+		setEmployeeToRemove(undefined);
+	};
+
+	useEffect(() => {
+		setModalIsVisible(!!employeeToRemove);
+		setShowRemovedEmployeeMessage(false);
+	}, [employeeToRemove]);
+
+	const removeEmployee = () => {
+		step7SetState({
+			...step7GetState,
+			completed: false,
+			employee:
+				step7State?.employee &&
+				step7State.employee.filter(
+					(employee) => employee?.id !== employeeToRemove?.id
+				),
+		});
+
+		setRemovedEmployee(
+			`${employeeToRemove?.firstName} ${employeeToRemove?.lastName}`
+		);
+		closeModal();
+
+		setTimeout(() => {
+			setShowRemovedEmployeeMessage(true);
+		}, 0);
+	};
+
+	const onSubmit: FormEventHandler<HTMLFormElement> = async (event) => {
+		event.preventDefault();
 
-	const onSubmit: SubmitHandler<Step7FormSchema> = async (data) => {
 		if (isSavingBeforeExiting) {
 			return;
 		}
+
+		if (step7State?.employee && step7State?.employee.length === 0) {
+			setShowRemovedEmployeeMessage(false);
+			setShowErrorMessage(true);
+			return;
+		}
+
 		await submitStep();
 		step7SetState({
-			...data,
+			...step7State,
 			completed: !isSavingBeforeExiting,
 			started: true,
 		});
 	};
 
 	return (
 		<FormContainer
+			formIntroduction="Add your employee details."
 			formTitle="Employees"
-			formIntroduction="Add details of any employees who will be handling food."
+			shouldFocusTitle={!showAddedEmployeeMessage}
 		>
-			<Stack as="form" gap={3} onSubmit={handleSubmit(onSubmit)} noValidate>
-				<Controller
-					control={control}
-					name="cuisine"
-					render={({ field: { ref, ...field } }) => (
-						<ComboboxMulti
-							label="TODO"
-							hint="TODO"
-							inputRef={ref}
-							options={cuisineOptions}
-							required
-							{...field}
-							id="cuisine"
-							invalid={Boolean(typeCorrectedErrors.cuisine?.message)}
-							message={typeCorrectedErrors.cuisine?.message}
-							maxWidth="xl"
-							block={false}
-						/>
-					)}
-				/>
-				<StepActions />
+			<Stack gap={2}>
+				{showErrorMessage && (
+					<PageAlert focusOnMount title="No employees added" tone="error">
+						<Text as="p">
+							You need to add some staff. You’re amazing, but you can’t do this
+							alone. Life is better with friends and team mates.
+						</Text>
+					</PageAlert>
+				)}
+				<H2 id="list-of-employees">List of employees</H2>
+				{showAddedEmployeeMessage && (
+					<SectionAlert
+						focusOnMount
+						onClose={() => setShowAddedEmployeeMessage(false)}
+						title={`${addedEmployee?.firstName} ${addedEmployee?.lastName} has been added as an
```

Comment:
These values could be `undefined`, right? So we could render `undefined undefined has been added as an employee`?

Reply:
Or is this just because of the `DeepPartial` we have?

Reply:
Yeah, I think it's the `DeepPartial`

Reply:
Yep, DeepPartial makes them possibly undefined
<img width="392" alt="image" src="https://github.com/user-attachments/assets/b0914bdc-f4fe-4652-95b9-2e93e1472b93">

Reply:
Actually the `DeepPartial` may be more accurate than I first thought now that we have clarity on the "Save and exit" flow. Any forms with "Save and exit" could have empty values even for required fields.

Reply:
Ah yeah!